cpp/src/rolling/rolling.cu

```diff
@@ -147,69 +288,73 @@ struct rolling_window_launcher
+         rmm::mr::device_memory_resource *mr,
+         cudaStream_t stream) {
 
-    rmm::device_scalar<size_type> device_valid_count{0, stream};
+      if (input.is_empty()) return empty_like(input);
 
-    if (input.has_nulls()) {
-        if (op == aggregation::COUNT) {
-            gpu_rolling<size_type, agg_op, op, block_size, true><<<grid.num_blocks, block_size, 0, stream>>>
-                (*input_device_view, *output_device_view, device_valid_count.data(),
-                 preceding_window_begin, following_window_begin, min_periods);
-        }
-        else {
-            gpu_rolling<InputType, agg_op, op, block_size, true><<<grid.num_blocks, block_size, 0, stream>>>
-                (*input_device_view, *output_device_view, device_valid_count.data(),
-                 preceding_window_begin, following_window_begin, min_periods);
-        }
-    } else {
-        if (op == aggregation::COUNT) {
-            gpu_rolling<size_type, agg_op, op, block_size, false><<<grid.num_blocks, block_size, 0, stream>>>
-                (*input_device_view, *output_device_view, device_valid_count.data(),
-                 preceding_window_begin, following_window_begin, min_periods);
-        }
-        else {
-            gpu_rolling<InputType, agg_op, op, block_size, false><<<grid.num_blocks, block_size, 0, stream>>>
-                (*input_device_view, *output_device_view, device_valid_count.data(),
-                 preceding_window_begin, following_window_begin, min_periods);
-        }
-    }
+      auto output = make_numeric_column(cudf::data_type{cudf::experimental::type_to_id<size_type>()},
+            input.size(), cudf::UNINITIALIZED, stream, mr);
 
-    output->set_null_count(output->size() - device_valid_count.value(stream));
+      cudf::mutable_column_view output_view = output->mutable_view();
 
-    // check the stream for debugging
-    CHECK_CUDA(stream);
+      // Passing the agg_op and aggregation::Kind as constant to group them in pair, else it
+      // evolves to error when try to use agg_op as compiler tries different combinations
+      if(op == aggregation::MIN) {
+          kernel_launcher<T, DeviceMin, aggregation::ARGMIN, WindowIterator, true>(input, output_view, preceding_window_begin,
+                  following_window_begin, min_periods, aggr, DeviceMin::template identity<T>(), stream);
+      } else if(op == aggregation::MAX) {
+          kernel_launcher<T, DeviceMax, aggregation::ARGMAX, WindowIterator, true>(input, output_view, preceding_window_begin,
+                  following_window_begin, min_periods, aggr, DeviceMax::template identity<T>(), stream);
+      } else {
+          kernel_launcher<T, DeviceCount, aggregation::COUNT, WindowIterator>(input, output_view, preceding_window_begin,
+                  following_window_begin, min_periods, aggr, string_view{}, stream);
+      }
 
-    cudf::nvtx::range_pop();
+      // If aggregation operation is MIN or MAX, then the output we got is a scatter map
```

Comment:
Do you mean gather map? (calls gather below, not scatter).

```suggestion
      // If aggregation operation is MIN or MAX, then the output we got is a gather map
```

Reply:
My mistake